packages/jest-config/src/index.ts

```diff
@@ -66,6 +66,27 @@ export async function readConfig(
   );
 
   const {globalConfig, projectConfig} = groupOptions(options);
+
+  // Process plugins after config is resolved
+  if (projectConfig.plugins && projectConfig.plugins.length > 0) {
```

Comment:
Can `projectConfig.plugins` be `null` or `undefined`? If yes, you can use optional chaining syntax `projectConfig.plugins?.length > 0`